### PR TITLE
[SeoBundle] Seo and social tab update

### PR DESCRIPTION
**Pull request by denbatte**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes
| Deprecations? | no
| Fixed tickets | /

This PR will:
- clean up the social and seo page tabs
- improve facebook og integration
- add twitter cards integration (including image and video cards)
- add google "in result search bar" functionality.

**BC**
The facebook_widget and linkedin widget twig extensions are removed. There are better ways to include them with far more flexibility.

Also, this update requires you to run migrations diff and migrate because some columns are removed and some are added.

### CODE REVIEW

File: src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/defaultsite/Resources/views/Pages/HomePage/view.html.twig [hunk 1]
```diff
@@ -11,3 +11,18 @@
...
+{% block footerscripts %}
+    <!-- Google search integration -->
+    <script type="application/ld+json">
+        {
+          "@context": "http://schema.org",
+          "@type": "WebSite",
+          "url": "{{ url("_slug", {url: '/'}) }}",
+          "potentialAction": {
+            "@type": "SearchAction",
+            "target": "{{ get_url_by_internal_name('search', app.request.locale) }}?query={search_term_string}",
```
Comment (wimvds) [inline src/Kunstmaan/GeneratorBundle/Resources/SensioGeneratorBundle/skeleton/defaultsite/Resources/views/Pages/HomePage/view.html.twig]: Not every site will have an active search page. You should only add this when a search page is actually available.

File: src/Kunstmaan/SeoBundle/Resources/views/Admin/Social/social.html.twig [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{% if resource.types is not empty %}
+    <fieldset class="panel panel-default form-properties">
+        <div class="panel-body">
+            {% for typeName in resource.types|keys %}
+                <ul class="nav nav-tabs" data-tabs="tabs">
+                    <li class="active"><a href="#og" data-toggle="tab">Open Graph</a></li>
```
Comment (wimvds) [inline src/Kunstmaan/SeoBundle/Resources/views/Admin/Social/social.html.twig]: No hardcoded copy in templates please, use translation keys

File: src/Kunstmaan/SeoBundle/Form/SeoType.php [hunk 3]
```diff
@@ -19,18 +20,52 @@ class SeoType extends AbstractType
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder->add('id', 'hidden')
-		->add('metaTitle', null, array(
-		    'label' => 'Meta title',
```
Comment (wimvds) [inline src/Kunstmaan/SeoBundle/Form/SeoType.php]: All labels, info_texts, placeholders, ... should use translation keys...

File: src/Kunstmaan/SeoBundle/EventListener/NodeListener.php [hunk 4]
```diff
@@ -44,6 +44,7 @@ public function adaptForm(AdaptFormEvent $event)
 
             $socialWidget = new FormWidget();
             $socialWidget->addType('social', new SocialType(), $seo);
+            $socialWidget->setTemplate('KunstmaanSeoBundle:Admin\Social:social.html.twig');
             $event->getTabPane()->addTab(new Tab('Social', $socialWidget));
```
Comment (wimvds) [inline src/Kunstmaan/SeoBundle/EventListener/NodeListener.php]: Please use a translation key for the tab name.

File: src/Kunstmaan/SeoBundle/Resources/views/SeoTwigExtension/metadata.html.twig [hunk 5]
```diff
@@ -2,38 +2,96 @@
...
 {% if seo.getOgImage() is defined and seo.getOgImage() is not null %}
+    {% if seo.ogImage.contentType == "remote/video" %}
+        {% if seo.ogImage.metaData.type == "youtube" %}
+            {% set videoUrl = "youtube.com/embed/"~seo.ogImage.metaData.code %}
+        {% elseif seo.ogImage.metaData.type == "dailymotion" %}
+            {% set videoUrl = "www.dailymotion.com/embed/video/"~seo.ogImage.metaData.code %}
+        {% else %}
+            {% set videoUrl = "player.vimeo.com/video/"~seo.ogImage.metaData.code %}
+        {% endif %}
+    <meta property="og:video" content="http://{{ videoUrl }}">
```
Comment (wimvds) [inline src/Kunstmaan/SeoBundle/Resources/views/SeoTwigExtension/metadata.html.twig]: Don't use hardcoded http references ... Either keep the protocol of the current request or omit the protocol

File: src/Kunstmaan/SeoBundle/Form/SeoType.php [hunk 6]
```diff
@@ -19,18 +20,52 @@ class SeoType extends AbstractType
...
+                'max_length' => 55,
+                'attr' => array(
+                'info_text' => 'The title tag is often used on search engine results pages. It should be less than 55 characters.'
+            )
+        ))
+        ->add('metaDescription', null, array('label' => 'Meta description', 'max_length' => 155));
+
+        $builder->add('metaRobots', 'choice', array(
+            'choices'   => array(
+                'noindex'       => 'No index',
```
Comment (kimausloos) [inline src/Kunstmaan/SeoBundle/Form/SeoType.php]: Try to use constants for these kind of lists

**Comment by yoshz** [comment]
What happened to keywords?

**Pull request merged**